tests/test_bug_features.py

```diff
@@ -266,3 +268,10 @@ def test_is_same_target_milestone(test_data, expected):
 @pytest.mark.parametrize("test_data, expected", FIRST_AFFECTED_PARAMS)
 def test_is_first_affected_same(test_data, expected):
     assert is_first_affected_same()(test_data) == expected
+
+
+def test_BugExtractor(feature_extractors, cleanup_functions):
+    BugExtractor([has_str(), has_url()], [fileref(), url()])
+    with pytest.raises(AssertionError):
+        BugExtractor([has_str(), has_str()], [fileref(), url()])
+        BugExtractor([has_str(), has_url()], [fileref(), fileref()])
```

Comment:
This way you are only testing the first call, as it raises the assertion and exits.

You should instead do:
```
with pytest.raises(AssertionError):
    BugExtractor([has_str(), has_str()], [fileref(), url()])


with pytest.raises(AssertionError):
    BugExtractor(...)
```